src/main/java/at/hannibal2/skyhanni/features/mining/MineshaftPityDisplay.kt

```diff
@@ -231,17 +231,19 @@ object MineshaftPityDisplay {
 
     init {
         RenderDisplayHelper(
-            condition = { isDisplayEnabled() },
+            condition = {
+                if (display.isEmpty()) {
+                    update()
+                    return@RenderDisplayHelper false
+                }
+                return@RenderDisplayHelper isDisplayEnabled()
+            },
             outsideInventory = true,
         ) {
-            // TODO dont do this logic in render, do this in tick
-            display.ifEmpty { update() }
-            if (display.isNotEmpty()) {
-                config.position.renderRenderables(
-                    listOf(Renderable.verticalContainer(display, 2)),
-                    posLabel = "Mineshaft Pity Display",
-                )
-            }
+            config.position.renderRenderables(
+                listOf(Renderable.verticalContainer(display, 2)),
```

Comment:
this should get moved into the spot where display gets created. (feel free to create a new todo instead of doing this in this pr)
listOf(Renderable.verticalContainer(display, 2)),

Reply:
good catch, done